keps/672-execution-policy/README.md

```diff
@@ -0,0 +1,383 @@
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+  - [Manage quota for Job sequence](#manage-quota-for-job-sequence)
+  - [Support complex DAGs with JobSet](#support-complex-dags-with-jobset)
+- [Alternatives](#alternatives)
+  - [Add ReplicatedJobsStatus parameter into the StartupPolicy API](#add-replicatedjobsstatus-parameter-into-the-startuppolicy-api)
+  <!-- /toc -->
+
+## Summary
+
+This KEP outlines the proposal to add the ExecutionPolicy API into JobSet. The ExecutionPolicy API
+allows to run sequence of ReplicatedJobs within a single JobSet.
+
+## Motivation
+
+Currently, JobSet supports the StartupPolicy API which allows to create Jobs in order after the
+first Job is in ready status. This can be useful when driver should be ready before workers.
+However, sometime high performance computing and machine learning users want to run sequence of
+ReplicatedJobs within JobSet. For example, it is common to run pre-processing,
+distributed fine-tuning, post-processing for LLM fine-tuning.
+
+### Goals
+
+- Add the ExecutionPolicy API for JobSet
+
+### Non-Goals
+
+- Support workflow management like DAGs outside of Job sequence.
+  - Users should consider to use Argo Workflows or Tekton Pipelines for such use-cases.
+- Allowing for a percentage of Jobs in a ReplicatedJob to be ready to consider the
+  whole ReplicatedJob to ready.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user, I want to fine-tune LLM using multi-node training with JobSet. I have the first
+ReplicatedJob for pre-trained model and dataset initialization, and the second ReplicatedJob for
+distributed fine-tuning.
+
+The example of JobSet looks as follows:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: fine-tune-llm
+spec:
+  executionPolicy:
+    executionPolicyOrder: InOrder
+    rules:
+      - targetReplicatedJobs:
+          - Initializer
+        replicatedJobsStatus: Succeeded
+  replicatedJobs:
+    - name: Initializer
+      template:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: model-initializer
+                  image: docker.io/kubeflow/model-initializer
+                  volumeMounts:
+                    - mountPath: /workspace/pre-trained-model
+                      name: model-initializer
+                - name: dataset-initializer
+                  image: docker.io/kubeflow/dataset-initializer
+                  volumeMounts:
+                    - mountPath: /workspace/dataset
+                      name: dataset-initializer
+              volumes:
+                - name: dataset-initializer
+                  persistentVolumeClaim:
+                    claimName: dataset-initializer
+                - name: model-initializer
+                  persistentVolumeClaim:
+                    claimName: model-initializer
+    - name: Node
```

Comment:
I think, we make the Node name here on purpose, since from the distributed training perspective this ReplicatedJob represents the single Training Node.
Here is the table which explains it: https://github.com/kubeflow/training-operator/tree/master/docs/proposals/2170-kubeflow-training-v2#worker-and-node-definition.
cc @tenzen-y

Reply:
We discussed today with @tenzen-y that we probably name this Job template as TrainerNode to make it clearer for user.